### PR TITLE
Updated sample applications,docker file and added support for build

### PR DESCRIPTION
**Pull request by rakeshsinghranchi**
-Updated sample applications
    -Output EXE Name to match dotnetapp-<<version>> format
-Added docker build file to build applications
-Updated Readme file.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
Since the readme has to be manually copied to DH, what do you think about having just one copy at the root?  Having three copies seems like a maintenance nightmare and has a high probability of them getting out of sync over time.

File: dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM microsoft/dotnet-framework:3.5
+
+ENV MSBUILD_VERSION 14.0
```
Comment (MichaelSimons) [inline dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build]: What depends on this?  Is it needed?  Same comment applies to 4.6.2.

File: dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM microsoft/dotnet-framework:3.5
+
+ENV MSBUILD_VERSION 14.0 
+ENV MSBUILD_DOWNLOAD_URL https://download.microsoft.com/download/E/E/D/EEDF18A8-4AED-4CE0-BEBE-70A83094FC5A/BuildTools_Full.exe 
+
+RUN powershell -NoProfile -Command \ 
+         $ErrorActionPreference = 'Stop'; \ 
+         Invoke-WebRequest %MSBUILD_DOWNLOAD_URL% -OutFile BuildTools_Full.exe; \ 
+         & BuildTools_Full.exe /Silent /Full
```
Comment (MichaelSimons) [inline dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build]: Should BuildTools_Full.exe be removed?  Same comment applies to 4.6.2.

**Comment by richlander** [comment]
LGTM

I'd like to follow the model I created in the dotnet-docker-samples repo -- https://github.com/dotnet/dotnet-docker-samples/tree/dockerhub:

- Short explanation at the root README.md w/a link to the other branch.
- Docker Hub README.md in the 4.6.2 directory.
- No README.md in the 3.5 directory.

I believe that this aligns with what @MichaelSimons is saying.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,84 @@
...
+
+The .NET Framework was first released by Microsoft in 2001. The latest version is [.NET Framework 4.6.2](https://www.microsoft.com/net/framework).
+
+> https://docs.microsoft.com/dotnet/articles/framework/
+
+![dotnet-icon](https://cloud.githubusercontent.com/assets/2608468/19951790/a0458278-a11d-11e6-86e4-660aaa22aa3c.png)
+
+# How to use these Images
+
+## Run a sample .NET Core application within a container
```
Comment (vivmishra) [inline README.md]: This needs to .NET application and not Core.

File: dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build (hunk 2 above)
Comment (MichaelSimons) [inline dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build]: @rakeshsinghranchi, @vivmishra  - I didn't see a response to this comment?  We talked about this in person, I thought this was going to be addressed?

Comment (rakeshsinghranchi) [inline dotnetapp-3.5/dotnetapp-3.5/Dockerfile.build]: Missed this one. Can you take care of this please?

**Pull request merged**